compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/generators/DelegatedMemberGenerator.kt

```diff
@@ -0,0 +1,260 @@
+            val irReturn = IrReturnImpl(startOffset, endOffset, irBuiltIns.nothingType, delegateFunction.symbol, irCall)
+            body.statements.add(irReturn)
+        }
+        delegateFunction.body = body
+        return delegateFunction
+    }
+
+    private fun generateDelegatedProperty(
+        subClass: IrClass,
+        irField: IrField,
+        superProperty: IrProperty,
+        firSuperProperty: FirProperty
+    ) {
+        val startOffset = irField.startOffset
+        val endOffset = irField.endOffset
+        val descriptor = WrappedPropertyDescriptor()
+        val modality = if (superProperty.modality == Modality.ABSTRACT) Modality.OPEN else superProperty.modality
+        symbolTable.declareProperty(
+            startOffset, endOffset,
+            IrDeclarationOrigin.DELEGATED_MEMBER, descriptor, superProperty.isDelegated
+        ) { symbol ->
+            IrPropertyImpl(
+                startOffset, endOffset, IrDeclarationOrigin.DELEGATED_MEMBER, symbol,
+                superProperty.name, superProperty.visibility,
+                modality,
+                isVar = superProperty.isVar,
+                isConst = superProperty.isConst,
+                isLateinit = superProperty.isLateinit,
+                isDelegated = superProperty.isDelegated,
+                isExternal = false,
+                isExpect = superProperty.isExpect,
+                isFakeOverride = false
+            ).apply {
+                descriptor.bind(this)
+                this.parent = subClass
+                getter = generateDelegatedFunction(subClass, irField, superProperty.getter!!, firSuperProperty.getter!!).apply {
+                    this.correspondingPropertySymbol = symbol
+                }
+                if (superProperty.isVar) {
+                    setter = generateDelegatedFunction(subClass, irField, superProperty.setter!!, firSuperProperty.setter!!).apply {
+                        this.correspondingPropertySymbol = symbol
+                    }
+                }
+                this.metadata = FirMetadataSource.Property(
+                    buildProperty {
+                        this.name = superProperty.name
+                        this.origin = FirDeclarationOrigin.Synthetic
+                        this.session = components.session
+                        this.status = FirDeclarationStatusImpl(firSuperProperty.status.visibility, modality)
+                        this.isLocal = firSuperProperty.isLocal
+                        this.returnTypeRef = firSuperProperty.returnTypeRef
+                        this.symbol = FirPropertySymbol(getCallableId(subClass, superProperty.name))
+                        this.isVar = firSuperProperty.isVar
+                    }
+                )
+                subClass.addMember(this)
+            }
+        }
+    }
+
+    private fun getCallableId(irClass: IrClass, name: Name): CallableId {
+        val classId = classifierStorage.findFirClass(irClass)?.classId
```

Comment:
You can take `classId` directly from `irClass` without using nasty `findFirClass`